core/src/main/java/org/apache/iceberg/rest/requests/UpdateNamespacePropertiesRequest.java

```diff
@@ -0,0 +1,125 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
+/**
+ * A REST request to set and/or remove properties on a namespace.
+ * <p>
+ * The request body specifies a list of properties to remove and a map
+ * of key value pairs to update.
+ * <p>
+ * Properties that are not in the request are not modified or removed by this call.
+ * Server implementations are not required to support namespace properties.
+ */
+public class UpdateNamespacePropertiesRequest {
+
+  private List<String> removals;
+  private Map<String, String> updates;
+
+  public UpdateNamespacePropertiesRequest() {
+    // Required for Jackson deserialization.
+  }
+
+  private UpdateNamespacePropertiesRequest(List<String> removals, Map<String, String> updates) {
+    this.removals = removals;
+    this.updates = updates;
+    validate();
+  }
+
+  UpdateNamespacePropertiesRequest validate() {
+    Preconditions.checkArgument((removals != null && !removals.isEmpty()) || (updates != null && !updates.isEmpty()),
+        "Cannot create a request to update a namespace's properties without specifying any props to update or remove");
+    return this;
+  }
+
+  public List<String> removals() {
+    return removals == null ? ImmutableList.of() : removals;
+  }
+
+  public Map<String, String> updates() {
+    return updates == null ? ImmutableMap.of() : updates;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("removals", removals)
+        .add("updates", updates)
+        .toString();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private final ImmutableSet.Builder<String> removalsBuilder = ImmutableSet.builder();
+    private final ImmutableMap.Builder<String, String> updatesBuilder = ImmutableMap.builder();
+
+    private Builder() {
+    }
+
+    public Builder remove(String removal) {
+      Preconditions.checkNotNull(removal,
+          "Cannot pass null key to remove to UpdateNamespacePropertiesRequest.Builder");
+      removalsBuilder.add(removal);
+      return this;
+    }
+
+    public Builder removeAll(Iterable<String> removals) {
+      Preconditions.checkNotNull(removals,
+          "Cannot pass null removals to UpdateNamespacePropertiesRequest.Builder");
```

Comment:
How about "Invalid list of properties to remove: null".

Reply:
I like that. i wasn't sure how to phrase it so thanks!

Reply:
Updated!